File: challenge1/solutions.py
# Python solutions to Bluprint Coding Challenge 1

# Solution 1
i = 378429294349777
stringint = str(i)
cumulative = 1
for digit in stringint:
  if digit == '0':
    continue
  else:
    cumulative = cumulative * int(digit)

print("The product of all digits in {0} is {1}".format(i,cumulative))

# Solution 2
def a(n):
  t=1
  for c in str(n):
    if int(c): t*=int(c)
  return t
  
# Solution 3
b=lambda n:reduce(lambda a,b: a*b,[int(c) for c in str(n) if int(c)])

File: challenge1/solutions.kt
// Kotlin Solutions to Bluprint Challenge 1

// Solution 1
fun challenge1a(n:Int): Int {
  var product = 1
  n.toString().forEach {c ->
    val i = c.toString().toInt()
    if (i > 0) product *= i
  }
  return product
}
// Solution 2
fun challenge1b(n:Int): Int {
  return n.toString().map({c->c.toString().toInt()}).filter({it>0}).fold(1) {s,i->s*i}
}
// Solution 3
fun challenge1c(n: Int): Int {
    return n.toString()
            .mapNotNull { char -> char.toString().toIntOrNull() }
            .filter { digit -> digit > 0 }
            .reduce(Int::times)
}
